### PR TITLE
[express-apollo-prisma] Unit tests for Queueing

### LINKED ISSUE #1099: [express-apollo-prisma] Unit tests for Queueing
Acceptance Criteria:

- [ ]  `/example-job` endpoint **handler** tested

### PR DESCRIPTION
**Pull request by kodejuice**
## Type of change

Closes #1099 

- [ ] Documentation change
- [ ] Bug fix

## Summary of change

Add tests for generator endpoint

## Checklist

- [ ] These changes follow the [contributing guidelines](https://github.com/thisdot/starter.dev/blob/main/CONTRIBUTING.md)
- [ ] This starter kit has been approved by the maintainers
- [ ] Changes for this new starter kit are being pushed to an integration branch instead of main
- [ ] All dependencies are version locked
- [x] This fix resolves #1099 
- [ ] I have verified the fix works and introduces no further errors

### CODE REVIEW

File: starters/express-apollo-prisma/src/queue/job-generator-handler.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import amqplib from 'amqplib';
+import { createJobGeneratorHandler } from './job-generator-handler';
+
+const MOCK_URL = 'amqp://localhost:5673';
+
+const SPY_AMQPLIB_CONNECT = jest.spyOn(amqplib, 'connect');
+
+describe('.createJobGeneratorHandler', () => {
+	let req: any;
```
Comment (iansamz) [inline starters/express-apollo-prisma/src/queue/job-generator-handler.spec.ts:9]: Req and Res can be imported from src/mocks/express. Please check out example in src\healthcheck\healthcheck-handler.spec.ts and src\graphql\server-context\server-context-middleware-options.spec.ts

File: starters/express-apollo-prisma/src/queue/job-generator-handler.spec.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			send: jest.fn(),
+			status: jest.fn().mockReturnThis(),
+		};
+		process.env = {
+			AMQP_URL: MOCK_URL,
+		};
+	});
+
+	afterAll(() => {
+		process.env = OLD_ENV;
```
Comment (iansamz) [inline starters/express-apollo-prisma/src/queue/job-generator-handler.spec.ts:31]: this throws error: should be ORIGINAL_ENV

File: starters/express-apollo-prisma/src/queue/job-generator-handler.spec.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		});
+	});
+
+	describe('invalid environment', () => {
+		beforeAll(() => {
+			process.env = {};
+		});
+
+		afterAll(() => {
+			process.env = OLD_ENV;
```
Comment (iansamz) [inline starters/express-apollo-prisma/src/queue/job-generator-handler.spec.ts:84]: this throws error: should be ORIGINAL_ENV

File: starters/express-apollo-prisma/src/queue/job-generator-handler.spec.ts (hunk 2 above)
Comment (kodejuice) [reply starters/express-apollo-prisma/src/queue/job-generator-handler.spec.ts:31]: yeah i know

Comment (kodejuice) [reply starters/express-apollo-prisma/src/queue/job-generator-handler.spec.ts:31]: i just commited the sugggestion

**Comment by kodejuice** [comment]
image

**Pull request merged**